ambry-store/src/main/java/com.github.ambry.store/DiskManager.java

```diff
@@ -111,6 +111,20 @@ public void run() {
           logger.error(
               "Could not start " + numFailures.get() + " out of " + stores.size() + " stores on the disk " + disk);
         }
+
+        ;
+        try {
+          List<DiskSpaceRequirements> requirementsList = new ArrayList<>();
+          for (BlobStore blobStore : stores.values()) {
+            DiskSpaceRequirements requirements = blobStore.getDiskSpaceRequirements();
+            if (requirements != null) {
+              requirementsList.add(requirements);
+            }
+          }
+          diskSpaceAllocator.initializePool(requirementsList);
+        } catch (StoreException e) {
+          logger.error("Error while starting DiskSpaceAllocator", e);
```

Comment:
It could still attempt to allocate, but it is safer to fail in this case, since an exception here probably means that we'll either have issues with disk space (preallocation failed), or that the pool became inaccessible/unmodifiable, in which case there will be problems returning files to the pool. 

I have changed this to fail all stores on the disk.